Randomize 'mains' (#23)
Workiva/test_html_builder: README.md

pub run build_runner test $(pub run test_html_builder:browser_aggregate_tests --mode=args [--release])
```

### Randomizing the browser aggregation file

> By default, the tests will be aggregating in a specific order and this will be maintained between
> each test run. In order to mimic the dart test runner's built in test shuffling capability a consumer can add
> this to their yaml file to randomly shuffle the test order.

Reviewer: I think we should start this off by explaining why this is necessary (in other words, why the test runner's randomization doesn't work with browser aggregation). Maybe something like this?

```suggestion
With `dart test`, you can randomize the order in which test files are run by using the `--test-randomize-ordering-seed` flag. However, when using the this package's browser aggregation feature, this method of test order randomization doesn't work because there will be only be one test file for each template. In other words, most of the tests will continue to be run in the same order.

For that reason, test_html_builder also supports a `randomize_ordering_seed` option that can be set in your `build.yaml`. It behaves essentially the same as the `--test-randomize-ordering-seed` flag, meaning that you can set the value to a specific seed or `"random"` if you'd like the seed to be picked at random for you.

Here's an example:
```

```yaml
targets:
  $default:
    builders:
      test_html_builder:
        options:
          ...

>           randomize_ordering_seed: random

Reviewer: We should include the `browser_aggregation: true` option here, too, since the randomization only applies to those aggregate tests. Speaking of which, we should probably warn if we detect that a package has configured a seed here but does not have browser aggregation enabled.

```

If you notice a test failure due to the specific shuffling that occurred during that test

> run, update the test_shuffle_seed to be whatever was output in the test runners log.

Reviewer: Oh this is a good point.. I didn't think about the fact that we'd want to be sure that the seed can be obtained from the logs for debugging purposes. Ok, maybe instead of logging the seed as info or warning, we could use `.shout(...)`? That should guarantee that it gets printed without it being misconstrued as a warning

Reviewer: ```suggestion
If you notice a test failure due to the specific shuffling that occurred during that test
run, update the `randomize_ordering_seed` option to be whatever was output in the test runners log.
```

### Note

> When randomizing the test order it is recommended to ignore the aggregated test file from version
> control to avoid constant updates.

Reviewer: ```suggestion
When randomizing the test order it is recommended to ignore the aggregated test files from version
control to avoid constant updates.
```
